source/adios2/operator/compress/CompressBlosc.h

```diff
@@ -60,6 +64,50 @@ class CompressBlosc : public Operator
                       const size_t sizeOut, Params &info) const final;
 
 private:
+    using bloscSize_t = int32_t;
+
+    /** Decompress chunked data */
+    size_t DecompressChunkedFormat(const void *bufferIn, const size_t sizeIn,
+                                   void *dataOut, const size_t sizeOut,
+                                   Params &info) const;
+
+    /** Decompress data written before ADIOS2 supported large variables larger
+     * 2GiB. */
+    size_t DecompressOldFormat(const void *bufferIn, const size_t sizeIn,
+                               void *dataOut, const size_t sizeOut,
+                               Params &info) const;
+
+    class __attribute__((packed)) DataHeader
```

Comment:
@pnorbert I fixed hopefully the `ubsan` issue in the CI by disallowing the compiler to add padding. 
The disadvantage is that there is no standard attribute in C++ :-(
Is this way ok? If not, do you have a suggestion on how I can fix this issue instead?

Reply:
if it helps, there is an [internal helper function](https://github.com/ornladios/ADIOS2/blob/master/source/adios2/helper/adiosMemory.h#L244) to add padding in a portable way.

Reply:
I am not sure if this helps. The problem is that maybe the ptr for the output is not aligned during the compression. If I use the helper and shift my data few bytes and later during decompress I get the data within a array with a different base address. If so it would be not possible to find the data because the required padding is different.

If I understand the ubsan error for the old code correctly and what I found in the net the packed attribute is required to guarantee that the struct has the same layout even if you compile blosc on a different system. This attributes forbits the compiler to transform the struct which is IMO correct.

I can also remove the header class und work with two pure integers but I found it better readable to use the header class because it allowes in the code to shift with `sizeOf(Header)` instead using a magic number 8byte.

I am not sure which solution guarantee portability.

Reply:
Maybe it is enough if I align the struct to 4 byte.
Is ADIOS guaranteeing that my pointers given to compress operators are 4 byte aligned?

Reply:
> The problem is that maybe the ptr for the output is not aligned during the compression. 

The function I shared can verify that returning the required padding in bytes (if 0 no padding is required).

> Is ADIOS guaranteeing that my pointers given to compress operators are 4 byte aligned?

That's a good question. It depends on the serialization buffer consuming this interface. The BP buffer is `char` so it shouldn't be a problem. I'd check with @pnorbert for other existing strategies.

> I am not sure which solution guarantees portability.

Yup, that's the challenge.

Reply:
It looks like Blosc is not covered by CI and thus breaking on MSVC: #2745